src/core-services/account/mutations/addAccountToGroup.js

```diff
@@ -47,58 +48,38 @@ export default async function addAccountToGroup(context, input) {
   const accountUser = await users.findOne({ _id: account.userId });
   if (!accountUser) throw new ReactionError("not-found", "No user found with that ID");
 
-  // Get a list of all the IDs of groups that belong to this shop
-  const allGroupsInShop = await Groups.find({
-    shopId
-  }, {
-    projection: {
-      _id: 1
-    }
-  }).toArray();
-  const allGroupIDsInShop = allGroupsInShop.map((grp) => grp._id);
+  const groupToAdd = account.groups.includes(groupId);
+  if (groupToAdd) throw new ReactionError("group-found", "Account is already in this group");
 
-  // Add all group roles to the user. Make sure this stays in this order.
-  // Remove former group roles before adding new group roles, in case some are in both.
+  // Get existing roles from a user
   const newAccountUserRoles = new Set((accountUser.roles || {})[shopId] || []);
 
-  const formerGroupId = (account.groups || []).find((grpId) => allGroupIDsInShop.indexOf(grpId) !== -1);
-  if (formerGroupId) {
-    const formerGroup = await Groups.findOne({
-      _id: formerGroupId
-    }, {
-      projection: {
-        permissions: 1
-      }
-    });
-    if (formerGroup) {
-      for (const formerRole of (formerGroup.permissions || [])) {
-        newAccountUserRoles.delete(formerRole);
-      }
-    }
-  }
-
-  for (const newRole of groupPermissions) {
-    newAccountUserRoles.add(newRole);
-  }
+  // Merge existing roles on user and roles from new group
+  // TODO(pod-auth): this will likely be removed in #6031, where we no longer get roles from user.roles
+  const newRoles = [...newAccountUserRoles, ...groupPermissions];
+  const uniqueRoles = _.uniq(newRoles);
 
+  // Add all group roles to the user. Make sure this stays in this order.
   const newAccountUserRolesArray = [...newAccountUserRoles];
   await ensureRoles(context, newAccountUserRolesArray);
```

Comment:
Seems like `const newAccountUserRolesArray =` line above should be deleted and `newRoles` should be passed into `ensureRoles` instead of `newAccountUserRolesArray`.